### DISCUSSION TITLE
ffprobe developer cannot be verified

### DISCUSSION THREAD

**Discussion by carla2908** [QutEcoacoustics/audio-analysis#560] [Q&A]
Hi! I am running the R script to install AP.exe on my Mac (https://ap.qut.ecoacoustics.info/guides/using_r), and when running the loop I get constant messages saying the '**_'ffprobe'' cannot be opened because the developer cannot be verified._** I keep canceling the window, but after it completes the run I just only see new folders in my output folder. Aren't I am suppose to expect the results of the analyses? Any feedback will be helpful. Thanks

**Comment by atruskie** [comment]
Hi there,

I haven't seen this exact problem before. Did you compile AP on your machine?

This is due to Apple's security policy. You would usually encounter this problem as soon as you run AP (and the problem would be AP itself). 

For reference, `ffprobe` is a program AP uses to get information about an audio file.

We wrote this article to help with these kinds of problems: https://openecoacoustics.org/resources/help-centre/software/unsigned/

Maybe try allowing whatever IDE you're using for R? 

Or the _Allow a single program_ section should work if you do it recursively for all the binaries in the [AP/audio-utils/osx-64/](https://github.com/QutEcoacoustics/audio-analysis/tree/master/lib/audio-utils/osx-x64/ffmpeg) folder, wherever it is installed on your computer.

**Comment by carla2908** [comment]
Hi! Thank you! Unfortunately, I have tried the three options shown in the link you sent but none of them work. That window still pops up. Do you have any other recommendation, please? Thanks again!

**Reply by atruskie** [reply]
Okay, last thing I can think of is to install ffmpeg and sox yourself (using Homebrew) and then to point AP to use those.

Once they're installed on your system, move or rename the audio-utils shipped with AP and then AP will find the ones installed on your computer.
